src/payment/strategies/mollie/mollie-payment-strategy.ts

```diff
@@ -0,0 +1,160 @@
+        const { payment , ...order} = payload;
+
+        if (!payment) {
+            throw new PaymentArgumentInvalidError([ 'payment' ]);
+        }
+
+        const { token, error } = await this._getMollieClient().createToken();
+
+        if (error) {
+            return Promise.reject(error);
+        }
+
+        return this._store
+                .dispatch(this._orderActionCreator.submitOrder(order, options))
+                .then(() => {
+                    return this._store.dispatch(this._paymentActionCreator.submitPayment({
+                        methodId: payment.methodId,
+                        paymentData: {
+                            formattedPayload: {
+                                credit_card_token: {
+                                    token,
+                                },
+                                browser_info: getBrowserInfo(),
+                            },
+                        },
+                    }));
+                })
+                .catch(error => this._processAdditionalAction(error));
+    }
+
+    finalize(): Promise<InternalCheckoutSelectors> {
+        return Promise.resolve(this._store.getState());
+    }
+
+    deinitialize(): Promise<InternalCheckoutSelectors> {
+        this._cardNumberElement?.unmount();
+        this._expiryDateElement?.unmount();
+        this._verificationCodeElement?.unmount();
+        this._cardHolderElement?.unmount();
+
+        return Promise.resolve(this._store.getState());
+    }
+
+    private _processAdditionalAction(error: any): Promise<InternalCheckoutSelectors> {
+        if (!(error instanceof RequestError) || !some(error.body.errors, {code: 'additional_action_required'})) {
+            return Promise.reject(error);
+        }
+
+        return new Promise(() => window.location.replace(error.body._links.checkoutUrl._links.checkoutUrl));
+    }
+
+    private _getInitializeOptions(): MolliePaymentInitializeOptions {
+        if (!this._initializeOptions) {
+            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+        }
+
+        return this._initializeOptions;
+    }
+
+    private async _loadMollieJs(merchantId: string, locale: string, testmode: boolean = false) {
+        if (this._mollieClient) {
+            return Promise.resolve(this._mollieClient);
+        }
+
+        return await this._mollieScriptLoader
+            .load(merchantId, locale, testmode);
+    }
+
+    private _getMollieClient(): MollieClient {
+        if (!this._mollieClient) {
+            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+        }
+
+        return this._mollieClient;
+    }
+
+    private _mountElements() {
+        const mollieClient = this._getMollieClient();
+        const { cardNumberId, cardCvcId, cardExpiryId, cardHolderId} = this._getInitializeOptions();
+
+        this._cardHolderElement = mollieClient.createComponent('cardHolder', {
```

Comment:
Can we send this options from the frontend?

Reply:
what do you mean from the frontend?

Reply:
you mean the component type?

Reply:
No, I meant the css classes.

Reply:
they are gone now